### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'player_features.p'

### ISSUE THREAD

**Issue by octaviodegodoy** [edouardthom/ATPBetting#5]
almost finishing compilation I got this error 
24200/24481 matches treated.
24300/24481 matches treated.
24400/24481 matches treated.
Traceback (most recent call last):
  File "main.py", line 83, in <module>
    features_player=load("player_features")

**Comment by clementine-per** [comment]
did you find a solution? I got the same error

**Comment by ricciolini** [comment]
Hi, anyone found a solution? I get the same error :(

**Comment by giambe99** [comment]
it's a problem related to the pickle stuff, just try to comment out from the main those 4 code line of
"dump" and those 4 of "load" that should work guys

**Comment by ricciolini** [comment]
@giambe99  thank you for the help, I will give it a go. 
By the way, have you tried it?
cheers

**Comment by giambe99** [comment]
Actually there are some problems with this code...
First of all it's clear that the elo feature is completly useless.
Also i suggest to delete about the first 50 lines and directly pd.read_csv() using the file atp_data that you can find on kaggle.

**Comment by giambe99** [comment]
You maight need to change datetime with datetime.datetime somwhere...
Anyway the code is usless if your scope is to earn money, I do strogly invite te author of this project to give it another shoot and improve it, cause actually its a very interesting topic

**Comment by giambe99** [comment]
@ricciolini sei italiano anche tu?
Are you skilled with xgboost?

**Comment by giambe99** [comment]
Admit we are here cause of kalle hallen video hahahaha

**Comment by ricciolini** [comment]
I'm a Python newbie ahah. I just wanted to try this code out for learning. There's a lot I don't understand, including xgboost.
And I'm Brazilian, with a bit of Italian blood :)
Thank's a lot for the tips, @giambe99  :)

**Comment by giambe99** [comment]
@ricciolini is an italian surname.
My advice 4 you is to first learn at least the basic logics of python, using pycharm i would say.
Then if you want to learn about ML xgboost is a powerfull tool wich you could run using anaconda.
Or if you just need to invest buy 10k of BTC and yo won't regreet that.
Bye boss

**Comment by ricciolini** [comment]
@giambe99 thanks for the advice :)
I'm with you on BTC. Got some already and looking into getting some more, finances permitting ;)
Take care and stay safe :)

**Comment by davide-creator** [comment]
> it's a problem related to the pickle stuff, just try to comment out from the main those 4 code line of
> "dump" and those 4 of "load" that should work guys

It didnt work for me. I have the same error. What can I do?

**Comment by Garagante** [comment]
> > it's a problem related to the pickle stuff, just try to comment out from the main those 4 code line of
> > "dump" and those 4 of "load" that should work guys
> 
> It didnt work for me. I have the same error. What can I do?

Was facing the same issue. For me, commenting the following lines (79 to 86 of main.py) worked : 
#dump(player_features,"player_features")
#dump(duo_features,"duo_features")
#dump(general_features,"general_features")
#dump(recent_features,"recent_features")
#features_player=load("player_features")
#features_duo=load("duo_features")
#features_general=load("general_features")
#features_recent=load("recent_features")

Let me know if you need me to check more 
Best regards

**Comment by adri567** [comment]
@giambe99 why you think the elo feature is completely useless?

**Comment by giambe99** [comment]
> @giambe99 why you think the elo feature is completely useless?
 
it's good to predict matches outcomes (with a certain precision),
but it's not good enought to predict outcomes better than the bookies...

for ex, elo 2000 vs elo 1600 ---> alomost certain, but u're not considering facts like recent injuries that could even leed to se a bookie offering a 1.80 1.90 

that said the best way to exploit bookies in a market is just to find the average of a given outcome and then bet on the platform who's offering the best deal (the best, not the profitable way).
The main problem is that the average rake is about 6-8%, this makes so hard to find a profitable strategies (almost impossible, tendent to 0).
if in coming years the market average rake will drop to something like 3% that would leed to more opportunities, but just if the current fluctuations between the bookies are kept ( a pretty unlikly scenario...)

**Comment by adri567** [comment]
> > @giambe99 why you think the elo feature is completely useless?
> 
> it's good to predict matches outcomes (with a certain precision),
> but it's not good enought to predict outcomes better than the bookies...
> 
> for ex, elo 2000 vs elo 1600 ---> alomost certain, but u're not considering facts like recent injuries that could even leed to se a bookie offering a 1.80 1.90
> 
> that said the best way to exploit bookies in a market is just to find the average of a given outcome and then bet on the platform who's offering the best deal (the best, not the profitable way).
> The main problem is that the average rake is about 6-8%, this makes so hard to find a profitable strategies (almost impossible, tendent to 0).
> if in coming years the market average rake will drop to something like 3% that would leed to more opportunities, but just if the current fluctuations between the bookies are kept ( a pretty unlikly scenario...)

Yeah, good point. I was thinking to look for an average of certain points without using the elo, because as you said sometimes it is useless. 
If I have some good predictions from the average points, I would give it a try on a platform.
